### ISSUE TITLE
can id in extended mode

### ISSUE THREAD

**Issue by veerusajjan** [cantools/cantools#626]
I have message id 0x9E384000 in the dbc file, but when I receive the message it is getting retrieved as 0x1E384000. is it correct message received ? is it extended id since MSB is set.? Please clarify on this.

**Comment by juleq** [comment]
Yes, bit 31 is the ext indicator in the dbc format. The id is only 29 billion wide. So what you receive is what was defined in your dbc.
